### PR TITLE
Save description for explorations

### LINKED ISSUE #1868: Description for DE queries
We want a DE query to have a description attribute that can be saved by the frontend.

- Add description attribute to `UIQuery` model
- Should be possible to get, create, patch with that
- add/run migrations

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #1868 

Added a description field in the UIQuery model which will allow storing description for each exploration. 

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
@Anish9901 I think a test for saving the description from the API would be helpful.

**Comment by Anish9901** [comment]
@kgodey working on it.

File: mathesar/models/query.py [hunk 1]
```diff
@@ -120,6 +120,12 @@ class UIQuery(BaseModel, Relation):
         unique=True,
     )
 
+    description = models.CharField(
+        max_length=280,
```
Comment (kgodey) [inline mathesar/models/query.py:124]: Why have a max length here? Why isn't it just a `TextField`?

Comment (Anish9901) [reply mathesar/models/query.py:124]: I thought of making it a `TextField` but then I thought users won't have descriptions that long for describing an exploration.

Comment (kgodey) [reply mathesar/models/query.py:124]: t think we should try to be consistent with table / schema descriptions, which are Postgres comments and can be arbitrarily long.

Comment (Anish9901) [reply mathesar/models/query.py:124]: Makes sense I'll change it to `TextField`.

File: mathesar/migrations/0007_auto_20221115_0627.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        migrations.AlterField(
+            model_name='uiquery',
+            name='initial_columns',
+            field=models.JSONField(validators=[mathesar.models.query._get_validator_for_list_of_dicts, mathesar.models.query._get_validator_for_initial_columns]),
+        ),
+        migrations.AlterField(
+            model_name='uiquery',
+            name='transformations',
+            field=models.JSONField(blank=True, null=True, validators=[mathesar.models.query._get_validator_for_list_of_dicts, mathesar.models.query._get_validator_for_transformations]),
+        ),
```
Comment (Anish9901) [inline mathesar/migrations/0007_auto_20221115_0627.py:33]: I don't know why this is getting added to the migration. It is similar to 0005 and 0006. and both 0005 and 0006 are actually the same, everytime I run `makemigrations` a new migration file is created with these fields.

Comment (mathemancer) [reply mathesar/migrations/0007_auto_20221115_0627.py:33]: For now, I think we should merge this with the extra migration steps.

Please make a bug ticket to figure out what's going on here.

**Review by mathemancer — APPROVED** [review]
Looks good, thanks!

**Pull request merged**